FireModule.lua

```diff
@@ -0,0 +1,64 @@
+--[[
+  Fire module as explained in SqueezeNet http://arxiv.org/pdf/1602.07360v1.pdf.
+--]]
+--FIXME works only for batches.
+
+local FireModule, Parent = torch.class('nn.FireModule', 'nn.Module')
+
+function FireModule:__init(nInputPlane, s1x1, e1x1, e3x3, activation)
+   self.nInputPlane = nInputPlane
+   self.s1x1 = s1x1
+   self.e1x1 = e1x1
+   self.e3x3 = e3x3
+   self.activation = activation or 'ReLU'
+
+   if self.s1x1 > (self.e1x1 + self.e3x3) then
+      print('Warning: <FireModule> s1x1 is recommended to be smaller'..
+            ' then e1x1+e3x3')
+   end
+   
+   self.module = nn.Sequential()
+   self.squeeze = nn.SpatialConvolution(nInputPlane, s1x1, 1, 1)
+   self.expand = nn.Concat(2)
+   self.expand:add(nn.SpatialConvolution(s1x1, e1x1, 1, 1))
+   self.expand:add(nn.SpatialConvolution(s1x1, e3x3, 3, 3, 1, 1, 1, 1))
+
+   -- Fire Module
+   self.module:add(self.squeeze)
+   self.module:add(nn[self.activation]())
+   self.module:add(self.expand)
+   self.module:add(nn[self.activation]())
+end
+
+function FireModule:updateOutput(input)
+   self.output = self.module:updateOutput(input)
+   return self.output
+end
+
+function FireModule:updateGradInput(input, gradOutput)
+   self.gradInput = self.module:updateGradInput(input, gradOutput)
+   return self.gradInput
+end
+
+function FireModule:accGradParameters(input, gradOutput)
+   self.module:accGradParameters(input, gradOutput)
+end
+
+function FireModule:float()
```

Comment:
Yeah will do it.
On Feb 27, 2016 11:39, "Nicholas Léonard" <notifications@github.com> wrote:

> In FireModule.lua
> <https://github.com/Element-Research/dpnn/pull/42#discussion_r54335493>:
>
> > +
> > +function FireModule:updateOutput(input)
> > +   self.output = self.module:updateOutput(input)
> > +   return self.output
> > +end
> > +
> > +function FireModule:updateGradInput(input, gradOutput)
> > +   self.gradInput = self.module:updateGradInput(input, gradOutput)
> > +   return self.gradInput
> > +end
> > +
> > +function FireModule:accGradParameters(input, gradOutput)
> > +   self.module:accGradParameters(input, gradOutput)
> > +end
> > +
> > +function FireModule:float()
>
> You can just over-ride the type() method instead of these 3.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Element-Research/dpnn/pull/42/files#r54335493>.
>